### DISCUSSION TITLE
Is there any way to log (or view) the SQL Query from Serenity.is?

### DISCUSSION THREAD

**Discussion by osdan-madrid** [serenity-is/Serenity#6384] [Help] [answered]
I've a problem with two tables, and I'd like to view if there is a join problem in the xyzRow.

Is there a way to see the query? maybe with a log ?

I want to see the T-SQL query generated in  xyzRow

**Comment by wezmag** [comment, marked as answer]
Let's assume you want to see the SQL query in ListRequestHandler.
You can add below code in the XyzListHandler class

```
protected override void OnBeforeExecuteQuery()
{
    base.OnBeforeExecuteQuery();
}
```
and then put a breakpoint on the closing curly bracket.

Hit F5 and start debugging.
Once it stops at the breakingpoint, use Watch window to examine `Query`
You will see the actual T-SQL query. 

![image](https://user-images.githubusercontent.com/340794/165463403-c0a6cc97-394a-492d-8a02-0ba0b3912bd9.png)

**Reply by osdan-madrid** [reply]
Thanks!!  😄

**Comment by stixoffire** [comment]
Enable debugging in the web config. App_data\log folder has logs.
